docs/src/testing.md

```diff
@@ -0,0 +1,73 @@
+# Testing
+
+All solvers use the tests in this repository as extra correctness tests for themselves.
+If we find a bug in one solver, instead of adding a test to that particular repository, we add it here so that all solvers can benefit.
+
+## Example of adding a test
+
+For this test, we care about correctness, rather than performance.
+We don't expect solvers to efficiently decide that they have already solved the problem, only that calling optimize! twice doesn't throw an error or give the wrong answer.
```

Comment:
This paragraph should go after the next one.

Reply:
done in PR #1189